### PR TITLE
`Forms`: Micro-app add enterprise login

### PR DESCRIPTION
**Pull request by kaushikrw**
### Summary of changes

- Added new login flow using the Authenticator toolkit component.
    - AGOL auto login using built-in credentials.
    - Enterprise login flow -> enter the portal url -> enter credentials -> login.
    - Skip sign-in will load any pre-specified portals using Anonymous connection.
- New navigation component to allow for the login flow. 
- Upgrades compose-navigation and compose to include animated navigation transitions. This also requires targetsdk to be 34.
- Re-designed `MapListScreen` to include a Profile view with the name of the user logged in and sign out action.
- Removed `PortalItemUseCase` since the layer information is no longer needed.

### CODE REVIEW

File: microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/screens/browse/MapListScreen.kt [hunk 1]
```diff
@@ -207,78 +234,145 @@ fun MapListItem(
...
+                            onClick = {
+                                expanded = false
+                                onRefresh(true)
+                            },
+                            leadingIcon = {
+                                Icon(imageVector = Icons.Default.Delete, contentDescription = null)
+                            }
+                        )
+                        DropdownMenuItem(
+                            text = { Text(text = "Sign Out") },
```
Comment (sorenoid) [inline microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/screens/browse/MapListScreen.kt:341]: Should this say `Sign In` when not signed in?

**Review by sorenoid — APPROVED** [review]
this looks great. It works well. I have some very minor feedback on menu items. I also have some ideas for improvements that can be done as new issues.

File: microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/data/network/ItemRemoteDataSource.kt [hunk 2]
```diff
@@ -30,32 +34,34 @@ class ItemRemoteDataSource(
...
+                    it.type == PortalItemType.WebMap
+                }.map {
+                    ItemData(it.url)
+                }
+            } else {
+                portalUserContent.items.filter {
+                    it.type == PortalItemType.WebMap
+                }.map {
+                    ItemData(it.url)
+                }
```
[the remark this answers is not in the document]
Comment (kaushikrw) [reply microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/data/network/ItemRemoteDataSource.kt:51]: Yes. If no folder is specified, all the webmaps from only the root are loaded. Should this be from root + all folders?

**Comment by kaushikrw** [comment]
Thanks @sorenoid, I will merge this and we can make new issue for enhancements like the [comment](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/pull/204#discussion_r1396000496) above.

**Pull request merged**